README.adoc

```diff
@@ -0,0 +1,180 @@
+= Documenting a Block
+
+== What you need to know
+
+* Knowledge of the block. What it does, how it is programmed, how do you interface to it,
+ and what are the configurable values.
+* How Scribble "Sections" work. Sections are the "generators" which produce documentation text.
+* The basics of Asciidoc. Asciidoc is the markup language used by Scribble.
+* The Jinja template processor. While complex Sections can be written
+ in Python, most Sections are more easily written as Jinja templates.
+
+== Sections of a Document
+
+Scribble organizes a document into Sections, where each Section produces
+a piece of text, and optionally invokes other Sections as subsections.
+When all of the Sections are evaluated, the resulting text is assembled into an AsciiDoc document.
+
+When creating documents for a new hardware block, we aren't creating a whole new document.
+Rather, we are creating special Sections (sometimes called "Snippets")
+which can be inserted into other documents.
+
+Scribble is expecting three Sections for each new hardware block:
+
+ - An "Overview"" paragraph describing the block.
+ - A "Programming" chapter describing how to use and program the block.
+ This section describes the registers, the memory mapped addresses,
+ and gives a detailed decription of how the block should be configured.
+ - A "HardwareInterface" chapter describing how to interface to the block.
+ This section describes the interrupts, clocks and signals used by the block.
+
+Typically, these sections will live in the "components" directory, in a subdirectory
+named after the block or device.
+
+For example, a new design for a "Whizbang" would have three files inside the following directories:
+```
+    - docs/scribble
+        - components
+            - Whizbang
+                - Overview.jinja2
+                - Programming.jinja2
+                - HardwareInterface.jinja2
+```
+
+=== What does a Section look like?
+
+A Section can be implemented as a Jinja template or as a Python function.
+In either case, it is invoked through Scribble's "Section" call.
+```
+Section("module name", scope, **context)
+```
```

Comment:
Let's both use the quadruple dash to indicate the code block, but let's also add the extra annotation so that it knows that this is a Python code block and adds syntax highlighting:

```
[source,python]
----
Section("module name", scope, **context)
----
```

Also, for the "module name" example, perhaps you should use a more realistic name? These have to match a qualified Python module path, so spaces aren't allowed, right?